Task dfl 2491 resource inspector improvements

p01 opened the pull request.

Commit:
first WIP of a Resource Manager organized by type of resources

Commit:
cleanup ResourceGroup handling, style tweaks, WIP handler to open individual resources

Commit:
added detail templates for all types of resources, moved the handlers out of the view into the ResourceManagerService, cleaned up OpenSingleResource and added a requestCreateRequest for resources in AppCache, added a few types in ResourceUtil

Commit:
added missing files

Commit:
fixed some contentLength/size/length mismatch, added .go_to_line in cls.ResourceDetailView.instance.createView

Commit:
clear_line_highlight resets the target_line

Commit:
removed my silly 'url' property, moved the parsing into its own method, and set the _is_parsed flag only when the uri is actually parsed

Commit:
ResourcePrototype now extends the URI class

Commit:
update the templates accordingly

Commit:
single click to open resources from the Documents service

Commit:
remove contextual menu to 'show resource'

Commit:
nicer rounding of the human readable file size, and added image/jpg MIME type

Commit:
add a getter for the extension of of a filename in the uri class

Commit:
tons of template updates, moved the opensingleresource into a more specific class

Commit:
':\'

Commit:
merge with mainline

Commit:
added missing setter in URI and fixed the _parse and _init calls

Commit:
merge bugfixes

Commit:
typo in uri

Commit:
commented out the group triggers, removed unnecessary name properties, added template for Flash resources

Commit:
fixed the search

Commit:
cleanup

Commit:
removed title attribute on the detail template

Commit:
Fix for CORE-46994 - Missing check for untracked frameID in cls.ResourceContext.update

Commit:
name the resource panels, removed dead resource panel

Commit:
switch to an absolute font size in the template for web Fonts preview

Commit:
Added disabled view ( necessary for the switch to Profiling mode )

Commit:
Let the TaskManager bind the callbacks. Added call to requestGetResource in _on_request_create_request

Commit:
Synched default with mainline

Commit:
Merge ../dragonfly-stp-1-resource-manager-work-converted

Commit:
Added missing reload-window class

Commit:
Fixed extension lookup in the type guessing of application/octet-stream resources

Commit:
Added missing reference to the network_details_templates

Commit:
minor style formatting + removed console.log(...)

Commit:
Fix for DFL-3326, Allow cells in the composite view to have their own min-width and min-height

Commit:
added min_width on the resource_tree_view

Commit:
WIP of DFL-3307, highlight cross origin resources

Commit:
Added a ResourceFrame class.
Make sure that resources loaded via the ResourceContext are bound to a frame.
Added test for upper domains in the sameOrigin check.

Commit:
Delete the frame and all its resrouces if the top resource of a frame is invalid.

Commit:
removed unnecessary method

Commit:
moved sameOrigin method to cls.ResourceUtil

Commit:
Make sure the frame still exists in the expand-collapse handler. It may have been invalidated since the view was last updated.

Commit:
Fix for DFL-3307, highlight cross origin resources. Added sameOrigin test for the children frames. TODO: better/cleaner styling

Commit:
Style cleanup + added .resource-different-origin class

Commit:
Highlight selected resource in the Resources tree

Commit:
Clean the context upon getting a top level frame.

Commit:
minor reformatting

Commit:
Fix for DFL-3342, redirected top level resource (of a frame) not supported

Commit:
Fix nowrap in the tree of Resources

Commit:
Merge branch 'master' into task_DFL_2491_Resource_Inspector_Improvements

Commit:
Merge branch 'master' into task_DFL_2491_Resource_Inspector_Improvements
df2 cleanrepo

Commit:
Early invalidation of orphan resources

Commit:
Better warning for orphan resources

Commit:
Merge branch 'task_DFL_2973_Make_Request_reviewed' into task_DFL_2491_Resource_Inspector_Improvements
ran cleanrepo

Commit:
Withstand missing reference URL in the sameOrigin check

Commit:
DRY cls.ResourceUtil.mime_to_content_mode and cls.ResourceUtil.type_to_content_mode

Commit:
Initial use of the Network API for the Resource Inspector. /!\ some things are not fully implemented yet, e.g. Detail view of a resource, default expand/collapse state of the tree, ...

Commit:
ResourceInfo now has an id ( aka resource_id ) and data ( aka responsebody ) which allow to show the detail view of a single resource

Commit:
Removed some dead code

Commit:
Cleaner get_resource(id) and get_resource_for_url(url)

Commit:
Removed unecessary ctx._resourceHash
Formatting

Commit:
Fixed _show_resource_by_id
Switch from cls.Resource to cls.ResourceInfo

Commit:
Merge branch 'task_DFL_2973_Make_Request_reviewed' into task_DFL_2491_Resource_Inspector_Improvements

Commit:
fullwidth zebra
default expand/collapsed ( branches deeper than the window > document are collapsed )

Commit:
initialized the resource_tee template only once

Commit:
Fix for DFL-3350 highlight selected resource in the tree

Commit:
reset selectedResourceID upon switching debugging context
no duplicate resource highligh

Commit:
Clean up of resource_util.js
Moved the guess_type() and get_meta_data() to resource_util.js
Use these in the cls.NetworkLoggerEntryPrototype

Commit:
Make sure there was at least one response
Expose the meta_data

Commit:
Fix the relative URL check in cls.ResourceDisplayBroker

Commit:
Check if this._show_resource failed.

Commit:
Fix for DFL-3570, Allow to show the same Resource in the tree of several Documents.

Commit:
Pruning of the lists of:
* WindowContext, keeping only the ones with saw_main_document set
* resources, keeping only the ones that are not unloaded.

Commit:
Dead code removal

Commit:
Throttle _listDocuments and make its callback pass a message type to the update method

Commit:
Syntax tweaks

Commit:
Sort the resources by ID in the tree ( avoids any jumping up and down because of asynch network events )

Commit:
Improved cls.ResourceRequest and how its used.
Deprecated cls.Resource in favor of cls.ResourceInfo
Clean up

Commit:
Broadcast messages when cls.ResourceRequest gets the resource_id and responsebody

Commit:
merged with task_DFL_2973_Make_Request_reviewed

Commit:
Added _reset() method and missing _suppressed_resource_update

Commit:
Added _suppressed_updates flag bouncing the updates while requesting the body of a resource

Commit:
Fix for DFL-3350, handling highlight of resources used by multiple runtimes.

Commit:
Fixed DFL-3278

Commit:
Renamed resource_test_view to resource_tree_view. Finally!

Commit:
Style update: better colors for the zebra table, hover and highlighting

Commit:
updated AUTHORS

Commit:
wrap the Resource tree in a DIV, fix the width issue with the full width zebra, tweak the color of the number of resources

Commit:
ran df2 cleanrepo

Commit:
Fix for DFL-3616 - Highlight resources with an "Error" HTTP Status

Commit:
preserve scrollTop position in the ResourceTree

Commit:
DFL-3195 Add token filter in the tree of resources

Commit:
Added some ui.strings

Commit:
Added ui_strings and changed a class name accordingly

Commit:
clean up the documentList mangling, including the pivotIDs and expandCollapse
moved the TYPE_GROUP_MAPPING to a const
replaced the last hardcoded strings with ui_strings

Commit:
Merge branch 'task_DFL_2973_Make_Request_reviewed' into task_DFL_2491_Resource_Inspector_Improvements

Commit:
Proper ellipsis

Commit:
remove some "debug" info.

Commit:
DFL-3575 discard closed windows, use the title from the WindowInfo in the resource tree

Commit:
Added ui_strings.S_RESOURCE_LOADING_PAGE

Commit:
added 'onbeforesearch' listener to filter the template

Commit:
Syntax fixes

Commit:
DRYed the _show_resource* code

Commit:
easier (to read) entry test for go_to_line(...)

Commit:
Merge branch 'task_DFL_2973_Make_Request' into task_DFL_2491_Resource_Inspector_Improvements

Conflicts:
	src/network/network_service.js

Commit:
Style tweaks

Commit:
Tidied the show_resource* methods. Passing the data along and handling String IDs.

Commit:
Switch to UID instead of ResourceInfo.id and the according clean up
Pass a base ResourceInfo object if available to RequestResource
Specifically suppress the "resource-update" for the URL requested

Commit:
Missing space between resource size and character encoding

Commit:
Merge branch 'bugfix_package_10' into cutting-edge

Commit:
Added cache-manifest mime type

Commit:
Handle base64 encoded text resources

Commit:
Improved the base64 detection ( less assignments )

Commit:
Fixed DFL-3642 - Clear detail view upon switching debugging context

Commit:
Fixed DFL-3644 - 	swap metadata ( image resolution & co ) and character encoding in the overview of the Resource Detail

Commit:
First phase of DFL-3645, Improved same_origin indication in the ResourceTree

Commit:
Switched from Search to Filter in the Resource tree

Commit:
Specified container class for the resource-detail-FOO-container with some padding

Commit:
No min-width for images, only a max-width:100%;
minor style tidy

Commit:
Show the URL of the resources in the inspection tooltip

Commit:
More switch from ResourceInfo.id to UID :\

Commit:
Improved style of the different origin indicator when the resource is highlighted

Commit:
Fix DFL-3646, Show full URL of the resource ( in the link ) in the Detail view

Commit:
Second phase of	DFL-3645, Improved same_origin indication in the ResourceTree.

Commit:
Preserve the scollLeft position

Commit:
syntax fixes

Commit:
DFL-3647, Expand the documents and groups that contain a resource matching the searchTerm of the token filter

Commit:
Fixed DFL-3648, Show match of the URL instead of the short_distinguisher when using the token filter in the resource tree

Commit:
Ran cleanrepo

Commit:
Added "font/x-woff" MIME type

Commit:
Clip the short distinguisher of the resources

Commit:
Added tooltip with the full URL of the documents

Commit:
Swapped the host and number of resources

Commit:
Preserve the original URL String

Commit:
Removed test that generated even the collapsed branches of the Resource tree

Commit:
Syntax & minor tests cleanup

Commit:
Only populate the list of the Document's resources once the document is set on the resource ( saves us from avoid the main resource of a document showing up in the parent document )

Commit:
Only render the ResourceTree view it if the template has changed.
Check if container.firstElementChild

Commit:
Lowered the THROTTLE_DELAY thanks to the previous optimization

Commit:
Moved the groupOrder to cls.ResourceTreeView and initialize all the default collapsed flags

Commit:
Merge branch 'master' into task_DFL_2491_Resource_Inspector_Improvements

Commit:
Recources style adjustments

Commit:
minor cleanup

hzr committed:
Fix DFL-3650 - need to escape control characters before parsing with innerHTML.

Commit:
Merge branch 'task_DFL-3533_resource_inspector_ux' into task_DFL_2491_Resource_Inspector_Improvements

Commit:
Removed the () around the count of resources in a subtree

hzr committed:
U+0009, U+000A and U+000D are all legal in XML.

Commit:
Removed the zebra background

Commit:
Tweak the resource count bubble

Commit:
Removed unecessary style

hzr committed:
Fix issue with inspection tooltip by showing the character with generated content instead of the escaped character.

Commit:
Prevent padding from causing horizontal scolling too early

Commit:
Fixed the network suppress_updates ( typo inception )

Commit:
Discard the 204 No content resources.

Commit:
Reset _loading flag upon switching debugging context

Commit:
Fixed scrolling issues with CODE resources

Commit:
Updating the icons

hzr committed:
Merge pull request #107 from operasoftware/escape-control-characters

Fix DFL-3650 - need to escape control characters before parsing with innerHTML

Commit:
Faux-colum^Wrow background for the ResourceDetail overview

Commit:
Simplified some CSS selectors

Commit:
whitespace

Commit:
Cleaner (destructured) check if we nee to fetch the list of documents + comments + removed dead variable

Commit:
Merge branch 'task_DFL-3533_resource_inspector_ux' into task_DFL_2491_Resource_Inspector_Improvements

Commit:
Removed dead mini template

Commit:
Keyboard shortcut to select the previous/next visible resource

Commit:
Locally attach the gradient background of the ResourceDetailView

Commit:
Bring back the "ShowResource" context menu

Commit:
Merge branch 'task_DFL_2973_Make_Request_reviewed' into task_DFL_2491_Resource_Inspector_Improvements

Commit:
Merge branch 'task_DFL_2973_Make_Request_reviewed' into task_DFL_2491_Resource_Inspector_Improvements

danfooo commented on src/client/client.js:
```diff
@@ -612,10 +612,19 @@ ui_framework.layouts.network_rough_layout =
-    width: 1000,
-    height: 1000,
-    children: [ { height: 1000, tabbar: { id: "resources", tabs: ['resource_all'] } } ]
+  dir:'h', width:1000,height:700,
```
Space after `:1000,`

p01 replied:
Fixed

danfooo commented on src/client/client.js:
```diff
@@ -612,10 +612,19 @@ ui_framework.layouts.network_rough_layout =
+  [
+    {
+      name: 'resource_tree_panel',
+      min_width:150,
```
Space after `min_width:`, will skip the following ;)

p01 replied:
Fixed

danfooo commented on src/lib/elementprototype.js:
```diff
@@ -172,6 +172,21 @@ Element.prototype.removeClass = function(cl)
+  {
+    this.addClass(name);
+  }
+}
```
Missing semicolon

hzr replied:
And might skip braces.

p01 replied:
Removed as this extension wasn't used anymore.

danfooo commented on src/lib/functionprototype.js:
```diff
@@ -0,0 +1,24 @@
+ * Throttle the function to run at most every delay ms
+ * @argument delay {Number}
+ */
+Function.prototype.throttle || (Function.prototype.throttle = function( delay )
```
Extra space around `delay`

p01 replied:
Fixed

danfooo commented on src/network/network_service.js:
```diff
@@ -581,7 +581,7 @@ cls.RequestContextPrototype = function()
 
   this.clear = function()
-  {
+    {
```
Looked better before, the next `};` too.

p01 replied:
Fixed

Commit:
Deleted the Metadata_drawer which is no longer used.

danfooo commented on src/network/network_service.js:
```diff
@@ -1067,14 +1067,10 @@ cls.NetworkLoggerEntryPrototype = function()
     // The current response is updated though, at the time it will be the correct one.
     // Multiple responses can get different types in this way.
-    if (!cls || !cls.ResourceUtil)
+    if (!cls || !cls.ResourceUtil || !cls.ResourceUtil.guess_type)
```
How could there be no `cls.ResourceUtil.guess_type`?

p01 replied:
Good point. I was overly cautious here.

p01 commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,445 @@
 ﻿window.templates = window.templates || {};
```
Maybe hold on reviewing this file. I can generate far less markup by flattening the structure. I have no seen any performance issue with the current markup but it doesn't hurt to generate less markup.

hzr commented on src/lib/functionprototype.js:
```diff
@@ -0,0 +1,24 @@
+ */
+Function.prototype.throttle || (Function.prototype.throttle = function( delay )
+{
+  var _callback = this;
```
No reason to use underscore here.

p01 replied:
Fixed

hzr commented on src/network/network_service.js:
```diff
@@ -1293,12 +1289,23 @@ cls.NetworkLoggerResponse.prototype = new cls.NetworkLoggerResponsePrototype();
-  this.id = entry.id;
+  this.responsecode = entry.current_responsecode;
+
+  var lastResponse = entry.requests_responses && entry.requests_responses.last;
```
last_response

p01 replied:
Fixed

hzr commented on src/network/network_templates.js:
```diff
@@ -228,7 +228,7 @@ templates.url_list_entry = function(selected, entry)
            templates.icon(entry),
            ["span",
-             (entry.short_distinguisher || entry.human_url).slice(0, 200),
+             (entry.short_distinguisher || entry.human_url || entry.url || "NO URL").slice(0, 200),
```
Lowercase "No", and this should be a UI string.

p01 replied:
Reverting to Daniel's code. It went through review already.

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+  this.createView = function(container)
+  {
+    if (this.resource && this.resource.data)
+      container.clearAndRender( templates.resource_detail.formatting_data(this.resource) );
```
No space around arguments.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+
+    container.clearAndRender( templates.resource_detail.update(this.resource) );
+    if(this.data)
+      this.go_to_line(container,this.data);
```
Space after the comma.

p01 replied:
Fixed

Commit:
Dead code removal

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+    container.clearAndRender(window.templates.disabled_view());
+  };
+
+  const HIGHLIGHTED_LINE_CLASSNAME = 'highlighted-line';
```
Use `var` instead of `const`. BTW, all new files should be written in strict mode, so add "use strict".

p01 replied:
Fixed.
Also switched my other .js files to strict mode.

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+  const HIGHLIGHTED_LINE_CLASSNAME = 'highlighted-line';
+  const RESOURCE_DETAIL_CONTAINER_CLASSNAME = 'resource-detail-container';
+  const TEXT = document.TEXT_NODE;
+  const ELE  = document.ELEMENT_NODE;
```
We don't usually line up assignments.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+        for (var pos = 0, len = value.length; pos < len; pos++)
+        {
+          var c = value.charAt(pos);
+          // Linefeed recognition will not support Acorn BBC spooled text output
```
Old comment that you didn't write, but I doubt this is an issue on the web. I would remove this comment.

p01 replied:
Removed.

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+              child.parentNode.normalize();
+              if (this._tops.length < 2)
+              {
+                this._target_line+=1;
```
Space around the "+=".

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+                var scroll_position = scroll_top + this._tops[0] - container_top;
+                if (scroll_position <= this._root_ele.parentNode.clientHeight)
+                {
+                  scroll_position-=64;
```
Space.

p01 replied:
Fixed

danfooo replied:
`-= 64`, maybe a const to annotate it?

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+                return;
+              }
+            }
+            if ((c == CR) && (value.charAt(pos+1) == LF))
```
Newline before this if statement.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+    this._line_found = false;
+    this._target_line = 0;
+    this._tops = [];
+    var _ele = container.querySelectorAll('.'+HIGHLIGHTED_LINE_CLASSNAME)[0];
```
Space around the "+". Won't comment on these kind of issues anymore, all should be fixed. :)

p01 replied:
Fixed

danfooo replied:
`.querySelector` seems like a better fit?

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+    if (this._root_ele)
+    {
+      this.clear_line_highlight(this._root_ele)
+      this._target_line = parseInt(data.lines[0]);
```
Use Number instead of parseInt.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+    if (!this._show_resource(resource, data))
+      this._show_resource_by_key(resource.uid, data);
+  };
+  this._show_resource_by_key = function(key, data)
```
Newline before.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+    var service = this._service;
+    var resource = service.get_resource(key) || service.get_resource_by_url(key);
+
+    var url = resource?resource.url:key;
```
Space around operators.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+    else
+      this._show_resource_by_key(key, data);
+
+    window.UI.instance.show_view( this.id );
```
No space around arguments.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+
+  var text_search = window.views.resource_detail_view.text_search = new TextSearch();
+
+  window.eventHandlers.input["resource-text-search"] = function(event, target)
```
event_handlers

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,256 @@
+      text_search.cleanup();
+  }
+
+  window.messages.addListener("view-created", on_view_created);
```
add_listener

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+  var THROTTLE_DELAY = 250;
+  var TYPE_GROUP_MAPPING =
+  {
+    'markup':ui_strings.S_HTTP_LABEL_FILTER_MARKUP,
```
Space after ":".

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+  this._network_logger = network_logger;
+
+
+  this._handle_listDocuments = function(status,msg)
```
list_documents. Space after ",".

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+
+  this._listDocuments = function()
+  {
+    this._tag_requestListDocuments = window.tagManager.set_callback(this, this._handle_listDocuments );
```
tag_manager

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+
+    // get list of window_contexts for which we saw the main_document
+    ctx.windowList = (this._network_logger.get_window_contexts()||[])
+    .filter(function(w)
```
I'd indent this.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+    {
+      // get all the (non-suppressed) resources with content
+      ctx.resourceList = (this._network_logger.get_resources()||[])
+      .filter(function(v)
```
Indent.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+      // augment the document objects,
+      // set the default collapsed flags
+      ctx.documentList  = this._documentList
+      .filter(function(d,i,a)
```
Indent.

p01 replied:
Fixed.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+          d.sameOrigin = cls.ResourceUtil.sameOrigin(p.url, d.url);
+
+          //  set the default collapsed flag
+          var hash = this. _collapsedHash;
```
Remove space.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+      // add group to each resource,
+      // sameOrigin flag to each resource
+      ctx.resourceList
+      .forEach(function(r)
```
Indent. or just move the forEach to the previous line.

p01 replied:
Indented for consistency' sake within this file.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+      var pivotIDs = [pivotID];
+      var collapsed = !hash[pivotID];
+
+      if (event.shiftKey)
```
The style guide says braces can be left out only if the statement is one physical line.

p01 replied:
Alrighty.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
-    this._current_context.update("urlfinished", data);
+    var e;
+    if (this._selectedResourceUID == uid)
+        return;
```
Indentation is two spaces.

p01 replied:
Fixed

hzr commented:
Before I continue, it would be nice if the code could be cleaned up a bit to conform to the style guide. Most issues are camelcase, extra spaces (or no spaces) around operators or inside arguments list.

p01 commented:
I can do a first pass of review fixes and code style cleanup.
Any recommendation to style the nested Array Iteration methods call ?

hzr commented:
You mean e.g.

``` javascript
  ctx.resourceList
  .forEach(function() {});
```

I'd do

``` javascript
  ctx.resourceList.forEach(function() {
    ...
  });
```

instead. If there are more, I'd do

``` javascript
  ctx.resourceList
    .forEach(function() {
    })
    .filter(function() {
    });
```

chriskr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,501 @@
+        pivotIDs.push.apply( pivotIDs, Object.keys(hash).filter( function(p)
+        {
+          return p.indexOf(pivotID+'_') == 0;
+        }));
```
We have prototype shortcuts, this could be written as `pivotIDs.extend(Object.keys(hash).filter(function(p){ return p.startswith(pivotID + "_"); }));`. 
(I'm not really reviewing and now I'm back on vacations.)

p01 replied:
Thanks!
Fixed.

p01 commented:
Yup.
Thanks.

Commit:
Review fixes

Commit:
Fixed hungry typos

danfooo commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,255 @@
+ * @extends ViewBase
+ */
+cls.ResourceDetailView = function(id, name, container_class, html, default_handler, network_logger) {
+  if (cls.ResourceDetailView.instance)
```
Didn't we give up on doing that? I think it can cause problems when enabling / disabling services? Not totally sure though.

hzr replied:
It should not be used as a singleton, no.

danfooo commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,255 @@
+  {
+    return cls.ResourceDetailView.instance;
+  }
+  cls.ResourceDetailView.instance = this;
```
It should not be called "service". We have changed that for the network logger too recently, a "service" is really only a scope service, like EcmascriptDebugger or the like.

p01 replied:
Gotta find another name then, because cls.ResourceManager is the generated class for the ResourceManager Scope Interface.

danfooo replied:
Maybe something like cls.ResourceManagerData

danfooo commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,255 @@
+
+  var HIGHLIGHTED_LINE_CLASSNAME = 'highlighted-line';
+  var RESOURCE_DETAIL_CONTAINER_CLASSNAME = 'resource-detail-container';
+  var TEXT = document.TEXT_NODE;
```
Should use double quotes throughout the file. We only have single quotes in legacy code (or right next to it).

p01 replied:
Fixed throughout the file.
These two lines are legacy code to me ;)

hzr commented on test-scripts/resource-tree-view/elementprototype.js:
```diff
@@ -0,0 +1,111 @@
+"use strict";
```
Just link this file from the src directory.

danfooo replied:
Is that just content for testing?

p01 replied:
Yes. It was an early prototype of Resource tree.
Not sure if I should keep it or remove it.

danfooo replied:
I guess it's confusing if it's in test-scripts but isn't really a test.

hzr commented on src/client/client.js:
```diff
@@ -612,10 +612,19 @@ ui_framework.layouts.network_rough_layout =
+    {
+      name: 'resource_tree_panel',
+      min_width: 150,
+      width: 300, tabbar: { id: 'resources_tree', tabs: ['resource_tree_view'], is_hidden: true }
```
No space inside object literals.

p01 replied:
I copied this syntax from:
ui_framework.layouts.js_rough_layout
ui_framework.layouts.profiler_rough_layout
and co

hzr replied:
Ok, that's old code though, but keep it for consistency.

p01 replied:
So... should I change anything here ? The more I look around `client.js` the more confused I am about the syntax :\

hzr replied:
No, just keep it like you have it here. `client.js` is old code and don't follow the style guide. We could clean it up in one go at some point.

hzr commented on src/lib/functionprototype.js:
```diff
@@ -0,0 +1,24 @@
+
+  return function()
+  {
+    if (last < Date.now()-delay)
```
Space around "-".

p01 replied:
Fixed

danfooo commented on src/resource-manager/resource_display_broker.js:
```diff
@@ -8,38 +10,34 @@ cls.ResourceDisplayBroker = function()
+      if (line)
+        data.lines = [line];
+
+      view.show_resource(resource, data);
```
Maybe merge with `show_resource_for_url` into `show_resource_for_id_or_url`, since they are the same?

p01 replied:
Good point.

danfooo commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,255 @@
+  this._span.textContent = ' ';
+  this._line_count = 0;
+  this._line_found = false;
+  this._target_line = 0;
```
Is this code new, or moved? `_tops`?

p01 replied:
/!\ This is legacy code.

Commit:
More review fixes

Commit:
Improved context adjustment of the line highlighting

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+
+  this._handle_list_documents = function(status, msg)
+  {
+    this._documentList = new cls.DocumentManager["1.0"].DocumentList(msg).documentList;
```
Maybe you haven't gotten around doing it yet, but this should be "document_list".

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+    });
+
+
+    this._update({type:'_handle_list_documents'});
```
Space after the colon.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+
+  this._populate_document_resources = function(r)
+  {
+    var documentID = r.document_id;
```
document_id. More of the same problem in this file, won't comment on it further.

p01 replied:
Fixed.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+    var documentID = r.document_id;
+
+    if (!this._documentResources[documentID])
+      this._documentResources[documentID]=[];
```
Space around "=".

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+    //  bounce if _suppress_updates
+    if (this._suppress_updates)
+    {
+      if (msg && msg.type=='resource-update')
```
Space around "==".

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+    {
+      if (msg && msg.type=='resource-update')
+      {
+        //  suppress the uid altogether if its URL matches the one we are requesting
```
Nitpick: extra space at the beginning of the comment. Same thing in some later comments.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+    ctx.groupOrder = this._view.get_group_order();
+
+    // get list of window_contexts for which we saw the main_document
+    ctx.windowList = (this._network_logger.get_window_contexts()||[])
```
Space around "||". There are similar problem further down, where there should be space around operators, won't comment on them further.

p01 replied:
Alright. I wasn't sure about spacing when using the `||` to provide a fallback value.
Fixed.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+
+      // mapping of the WindowIDs in the debugging context
+      var windowID_index = {};
+      ctx.windowList.forEach(function(w, i){ windowID_index[w.id] = i; });
```
Space before the brace.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+              nullDocumentID = true;
+
+            if (d.resourceID != null)
+              ctx.documentResourceHash[ d.resourceID ] = d.documentID;
```
No space inside brackets, same problem a few lines down.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+
+    if (this._selectedResourceUID)
+    {
+        e = document.querySelector('*[data-resource-uid="'+ this._selectedResourceUID +'"]');
```
This is indented two extra spaces.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+    if (this._context)
+      this._context.selectedResourceUID = uid;
+
+    e = document.querySelector('*[data-resource-uid="'+ this._selectedResourceUID +'"]');
```
The "*" is unnecessary.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+      uid = list[ (list.length+pos+increment) % list.length ];
 
-  this._on_debug_context_selected_bound = function()
+    this.highlight_resource( uid );
```
Extra spaces around the argument.

p01 replied:
Fixed

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+      var RESOURCE_ID = 0;
+      this.resource_id = message[RESOURCE_ID];
+
+        //  broadcast that we are working on the resource this.resource_id
```
Extra space before the comment.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+      var resourceData = new  cls.ResourceManager["1.2"].ResourceData(message);
+      if (this._retries == MAX_RETRIES || resourceData.content)
+      {
+
```
Extra newline.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+      {
+
+        // content -> mock a cls.NetworkLoggerEntry and instanciate a cls.ResourceInfo
+        this.requests_responses = [{responsebody:resourceData}];
```
Space after colon.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
-        this.human_url = this.type + " data URI";
+        // resource of known type-> call with appropriate transport mode.
+        var response_type = cls.ResourceUtil.type_to_content_mode(this.type);
+        var transport_type = response_type=='datauri'?TRANSPORT_DATA_URI:TRANSPORT_STRING;
```
Missing whitespace.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,496 @@
+        var resourceInfo = new cls.ResourceInfo(this);
+        if (!this.resourceInfo)
+            this.resourceInfo = resourceInfo;
+          else
```
Wrong alignment.

hzr commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+.resource-detail-container-empty
+{
+  height: 100%;
+  width:100%;
```
Missing space.

hzr commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+.resource-detail-container
+{
+  background-image: linear-gradient(180deg, #eee 54px, #e9e9e9 55px);
+  background-size:100% 55px;
```
Missing space.

hzr commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+  line-height: 16px;
+  border-radius: 3px;
+  padding: 0px 5px;
+  margin-left:4px;
```
Space

hzr commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+  border-radius: 3px;
+  padding: 0px 5px;
+  margin-left:4px;
+  vertical-align:middle;
```
Space

hzr commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+	DEPTH_IDENTATION: 18,
+	DISTINGUISHER_MAX_LENGTH: 64,
+
+	_get_short_distinguisher:function(url)
```
Space after colon.

hzr commented on src/resource-manager/resource_tree_view.js:
```diff
@@ -0,0 +1,216 @@
+  //	const
+  var THROTTLE_DELAY = 250;
+  var GROUP_ORDER =
+  [
```
Move the bracket up on the previous line.

hzr commented on src/resource-manager/resource_tree_view.js:
```diff
@@ -0,0 +1,216 @@
+    ui_strings.S_HTTP_LABEL_FILTER_OTHER
+  ];
+
+    // "private"
```
Space before the comment, but the comment is unnecessary.

Commit:
Review fixes

Commit:
DFL-3639: URL line in Storage tab is cut off

Commit:
Improved String descriptions

Commit:
Merge pull request #111 from operasoftware/danfooo_bugfixes

DFL-3639: URL line in Storage tab is cut off

danfooo commented on src/resource-manager/resource_display_broker.js:
```diff
@@ -49,10 +47,24 @@ cls.ResourceDisplayBroker = function()
+      this.show_resource_for_id(id, line);
+    else if (url)
+    {
+      //  resolve the URL based on that of the runtime if we only have a relative path
```
Space

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
+    "*": ui_strings.S_HTTP_LABEL_FILTER_OTHER
+  };
+
+
```
One newline should do, further down too.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
+    });
+
+
+    this._update({type: "_handle_list_documents"});
```
Seems the faked message argument won't really be looked at, maybe skip it?

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
+      {
+        // suppress the uid altogether if its URL matches the one we are requesting
+        var r = this._network_logger.get_resources([msg.id]);
+        if (r && r[0] && r[0].url == this._suppress_updates_url)
```
`r` will always be at least an empty array here, so you could do `var r = this._network_logger.get_resources([msg.id])[0];` and then check if you have it and so on.

p01 replied:
Correct!

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
-    if (data.parentDocumentID) { return; }
-    this._current_context = new cls.ResourceContext();
+    // bounce if _suppress_updates
+    if (this._suppress_updates)
```
The suppressing is quite complex, maybe you could skip the `_suppress_updates` bool and only use `_suppress_updates_url`?

p01 replied:
Good point . `_suppress_updates_url` would be enough.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
+        if (r && r[0] && r[0].url == this._suppress_updates_url)
+          this._suppress_uids[msg.id] = true;
+      }
+      return setTimeout(this._update_bound, THROTTLE_DELAY);
```
So for a while all updates are avoided? Is that for when you don't know the to-be-blocked uid yet? Probably explains why you need "this._suppress_updates", okay then.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
+    var ctx = {};
+
+    // get the order of the groups of resources
+    ctx.groupOrder = this._view.get_group_order();
```
`group_order`

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
+    ctx.windowList = (this._network_logger.get_window_contexts() || [])
+      .filter(function(w) {
+        return w.saw_main_document;
+      });
```
I'm not sure about the various inline filter functions here, update get's called a lot, right? It probably won't make much difference but we normally avoid defining them inline then. You can use helpers.eq in some cases, maybe add a helper for this simple "has a truthy prop by name x" case here.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
+      // get all the (non-suppressed) resources with content
+      ctx.resourceList = (this._network_logger.get_resources() || [])
+        .filter(function(v) {
+          return !this._suppress_uids.hasOwnProperty(v.uid) && v.responsecode != 204;
```
Maybe make a `var CONTENT_LESS_RESPONSE = 204`. Have you tried if you actually get resources of that sort from the network logger?

p01 replied:
I did get a 204 resource from the network_logger. IIRC it was for a URI of some stats/tracking service.

Trying to display such resource is fun. It goes through all the normal netwok/resource Scope API calls and eventually falls back to a window.open(url) which is also empty since the resource has no content. Good times.

hzr replied:
I would not make a constant for that, IMO it would be harder to read. 204 is not magic in this context.

danfooo replied:
But that will happen with any other empty resource? The special casing is good then, one case less where things will go wrong. That rough fallback is  kind of temporary too.

Ok about the const.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,492 @@
+          return !this._suppress_uids.hasOwnProperty(v.uid) && v.responsecode != 204;
+        }, this);
+
+      ctx.documentResourceHash = {};
```
Shouldn't CamelCase.

Commit:
Review fixes including _suppress_updates_url alone to suppress the "resource-update" of the resource data being fetched.

danfooo commented on src/build-application/build_resource_manager_1_0.js:
```diff
@@ -3,9 +3,20 @@
-  //new cls.ResourceManagerFontView('resource_fonts', "Fonts", 'scroll', '', '');
-  //new cls.ResourceManagerImageView('resource_images', "Images", 'scroll', '', '');
+  new cls.ResourceTreeView("resource_tree_view",
+                           "Tree View",
```
I'm not sure about the "name", seems it's only used for settings and to display on the subtabs and temporary tabs, and none of that applies to your views? Maybe it should be empty string then, "Detail" too. They are also very generic and could collide when they do get settings.

Commit:
Fix for DFL-3661, New resource tree view should have enough min-width to never cut off filter field
( Based on the 2px and 1px padding+border of the <toolbar> and 166px width of the <toolbar-filters> )

danfooo commented on src/network/network_service.js:
```diff
@@ -1293,12 +1289,23 @@ cls.NetworkLoggerResponse.prototype = new cls.NetworkLoggerResponsePrototype();
+  {
+    this.data = last_response.responsebody;
+    if (cls.ResourceUtil && cls.ResourceUtil.get_meta_data)
+      this.data.meta = cls.ResourceUtil.get_meta_data(this);
```
Maybe this should be done on the entry, once a responsebody is set. Making a `new Image()` per image resource whenever you call `get_resources` seems a bit wasteful, this is probably called a lot.

p01 replied:
True. A `getter`, like in the URI class, could work too. What do you think ?

danfooo replied:
It would need to be on the response, then it could be cached in the getter, at least as soon as it has a responsebody.
Sidenote, I thought unfinished / body-less entries wouldn't make a ResourceInfo, but it seems that's not true?

danfooo commented on src/resource-manager/resource_util.js:
```diff
@@ -155,80 +154,69 @@ cls.ResourceUtil.type_to_string_map = {
   {
-    return null;
+    var i=new Image();
+    i.src=data.content.stringData;
```
Space before and after `=`, possibly in more lines

danfooo commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,265 @@
+    this._tops = [];
+    var _ele = container.querySelector("." + HIGHLIGHTED_LINE_CLASSNAME);
+    if (_ele)
+      _ele.removeClass(HIGHLIGHTED_LINE_CLASSNAME)
```
Missing semicolon

Commit:
Added a note about throttled methods.

danfooo commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,265 @@
+{
+  new ToolbarConfig(
+  {
+    view:"resource_detail_view",
```
Space after `:`

danfooo commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,265 @@
+    get_instance().
+    get_global_handler().
+      register_shortcut_listener
+      (
```
Pretty crazy style, `(` should go on the same line, why indent `register_shortcut_listener`?

danfooo commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,265 @@
+  {
+    if (msg.id === "resource_detail_view")
+    {
+      var scroll_container = msg.container.querySelector(".request-details");
```
Can't find that classname referenced anywhere. Isn't it always `.resource-detail-container` being scrolled?

Commit:
Review fixes

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
 {
   if (cls.ResourceManagerService.instance)
-  {
     return cls.ResourceManagerService.instance;
```
Still singleton?

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+      var null_document_id = false;
+      var document_id_index = {};
+      // filter the documentId that belong in the windowIdList,
+      // set null_document_id flag,
```
Extra space. And in general there is lots of camelCasing in here.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+      return;
+
+    if (this._selected_resource_uid)
+    {
```
Should use your own `removeClass`

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+      return;
+
+    var parent = target.get_ancestor('[data-resource-uid]');
+    if (!parent)
```
Doesn't that need a check for null?

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+    var uid;
+    var list = this._context.visibleResources;
+    var pos = list.indexOf(this._selected_resource_uid);
+    if (pos == -1)
```
Shouldn't flip to first after last, other views don't either.

danfooo replied:
Should also scroll the newly selected into view. May be easier to find the next/prev on the DOM instead, see `network-view.js:300`.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
-    var filterfun = function(e) { return types.indexOf(e.type) > -1;};
-    return this.resources.filter(filterfun);
+    var ctx = this._context;
+    if (!ctx)
```
Using `helpers.eq` would get you caching.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
-    return this.resources.filter(filterfun);
+    var ctx = this._context;
+    if (!ctx)
+        return null;
```
No need to check for `resource`

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+
+    var resource = ctx.resourceList.filter(function(v) { return v.uid == uid; });
+    return resource && resource.last;
   };
```
Sooo similar to `this.get_resource`

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+    this.url = url;
+    this.resourceInfo = resourceInfo;
+    this._calback_data = data;
+    this._callback = callback;
```
Hm, `window.tag_manager` is not that much longer, to store it makes it loom like it might be something else.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+    this._calback_data = data;
+    this._callback = callback;
+    this._retries = 0;
+
```
Are the checks so it works in tests?

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+    this.resourceInfo = resourceInfo;
+    this._calback_data = data;
+    this._callback = callback;
+    this._retries = 0;
```
Would prefer `_resource_manager_service`, because we often confuse our own data handling and the actual services.

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
-    }
+    this.url = url;
+    this.resourceInfo = resourceInfo;
+    this._calback_data = data;
```
Typo: `_callback_data`

danfooo commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,487 @@
+
+  this._on_request_get_resource = function(status, message)
+  {
+    if (status == SUCCESS)
```
How come `this._retries == MAX_RETRIES` doesn't mean we should fallback?

Commit:
More review fixes

danfooo commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+  width: 250px;
+}
+
+.resource-tree-expand-collapse.close > ul
```
Double class selector should be avoided

danfooo commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+}
+.resource-detail-container
+{
+  background-image: linear-gradient(180deg, #eee 54px, #e9e9e9 55px);
```
We normally support a few client versions back, so should include prefixed.

danfooo commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+  top: 55px;
+  left: 0px;
+  right: 0px;
+  bottom: 0px;
```
`0` without `px`, above too.

danfooo commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+  color: #d2001c;
+}
+
+a.external
```
Such generic stuff shouldn't be in `resource_style.css`

danfooo commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
 
-.resource-manager td:nth-child(1) {
-  width: 16px;
+/* zebra resources */
```
Really?

danfooo commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,310 @@
+  display: inline-block;
+  line-height: 16px;
+  border-radius: 3px;
+  padding: 0px 5px;
```
`0 5px`

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+	_get_short_distinguisher: function(url)
+	{
+		var name = url.filename || url.short_distinguisher || url;
+		if (name.length>this.DISTINGUISHER_MAX_LENGTH)
```
Whitespace around operator, many more in this file.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+templates.resource_tree =
+{
+	URL_MATCH_CONTEXT_SIZE: 10,
+	DEPTH_IDENTATION: 18,
```
Should that be `DEPTH_INDENTATION`?

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+	{
+		var name = url.filename || url.short_distinguisher || url;
+		if (name.length>this.DISTINGUISHER_MAX_LENGTH)
+			name = name.slice(0, this.DISTINGUISHER_MAX_LENGTH)+'…';
```
Could `text-overflow: ellipsis;` be used instead?

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+	{
+		var hash = context.collapsed;
+
+		//	expand all pivots when searching
```
Extra whitespace

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+		var hash = context.collapsed;
+
+		//	expand all pivots when searching
+		if (context.searchTerm != '')
```
Should use double quotes as primary, throughout the file.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+		return { collapsed:collapsed, tpl:tpl };
+	},
+
+	update:function(context)
```
Whitespace after `:`, throughout the file.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+
+	window:function(context, w)
+	{
+		var windowInfo = window.window_manager_data.get_window(w.id);
```
window_info

Commit:
Review fixes + flat template for the resource_tree

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,447 @@
+	windows:function(context)
+	{
+		var tpl =
+			['div',
```
Can go to previous line, indent one less for the next lines.

Commit:
Look for the appropriate runtime ( and window_id ) in the resource_display_broker

Commit:
Look for the appropriate runtime ( and window_id ) in the resource_display_broker

Commit:
Took back fixes to english strings, resulting from DFL-3602, matches english.db rev 4fd3aa1

Commit:
Ran cleanrepo

Commit:
ResourceRequest - minor clean up and fallback data object

Commit:
Style cleanup

Commit:
No more ResourceManager singleton.

Commit:
Improved keyboard navigation

Commit:
resolving criss-cross commits

Commit:
More review fixes

Commit:
Review fixes + word-break and word-wrap to the Font preview

Commit:
Minor clean up and comments

Commit:
Semi and extra newline

hzr commented on src/resource-manager/resource_detail_view.js:
```diff
@@ -0,0 +1,254 @@
+    this._line_found = false;
+    this._target_line = 0;
+    this._tops = [];
+    var _ele = container.querySelector("." + HIGHLIGHTED_LINE_CLASSNAME);
```
No reason to use underscore here.

hzr commented on src/resource-manager/resource_display_broker.js:
```diff
@@ -1,63 +1,63 @@
+      id_or_url = url;
+    }
+
+    if(line)
```
Space after "if".

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,508 @@
+          r.document_id = document_id;
+
+        var d = this._document_list[document_id_index[r.document_id]];
+        if(!d)
```
Space.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,508 @@
-    }
-  }
-}
+    return this._get_resource_by_key_value('uid', uid);
```
Quotation marks instead of apostrophes, same below.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,508 @@
-    }
+    this.url = url;
+    this.resource_info = resource_info;
+    this._data = data||{};
```
Missing spaces.

hzr commented on src/resource-manager/resource_service.js:
```diff
@@ -1,285 +1,508 @@
+        var resource_info = new cls.ResourceInfo(this);
+        if (!this.resource_info)
+            this.resource_info = resource_info;
+          else
```
Wrong indentation.

hzr commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,314 @@
-  width: 250px;
+.resource-tree li > h2:hover
+{
+  background-color: rgba(137, 173, 232, 0.196);
```
alpha: 0.2

hzr commented on src/resource-manager/resource_style.css:
```diff
@@ -136,39 +121,314 @@
+.resource-highlight
+{
+  color: #fff;
+  background-color: rgba(55, 115, 211, 0.702);
```
alpha: 0.7

hzr commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,421 @@
+	{
+		var document_resources = context.document_resources[d.documentID] || [];
+		var resources = context.resources.filter(function(r) {
+				return document_resources.contains(r.uid);
```
Wrong indentation.

hzr commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,421 @@
+	this.resource = function(context, depth, r)
+	{
+		var search = context.search_term;
+		var partial_URL_match = "";
```
url

hzr commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,421 @@
+			["span",
+				(is_error ? info.response_code + " - " : "") +
+				ui_strings.S_RESOURCE_SENT_AND_GUESSED_TYPE
+				.replace("%(SENT)s", resource.data.mimeType)
```
I'd indent this and the following line two spaces.

p01 replied:
Just the `.replace(...)` calls ? Should the `(info.chara...`part have the same indentation as the `.replace(...)` or as the lines before them ?

hzr commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,421 @@
+	{
+		var data = resource.data.content.stringData;
+		var pos = data.indexOf(",");
+		var is_base64 = data.lastIndexOf(";base64", pos) != -1;
```
Why lastIndexOf instead of indexOf?

p01 replied:
This is a speed optimization.
`StringData` is a data:uri, so it can a _very_ long string. It may not be in base64, so by using `pos` as fromIndex of a lastIndexOf call we limit the search to a few dozen characters instead of potentially millions of characters.

hzr commented on src/resource-manager/resource_tree_view.js:
```diff
@@ -0,0 +1,214 @@
+    ui_strings.S_HTTP_LABEL_FILTER_OTHER
+  ];
+
+  // "private"
```
Comment not necessary.

hzr commented on src/resource-manager/resource_tree_view.js:
```diff
@@ -0,0 +1,214 @@
+  {
+    var ctx = this.service.get_resource_context();
+    var target = container.firstElementChild;
+    var scroll_top = target?target.scrollTop:0;
```
Missing whitespace.

hzr commented on src/resource-manager/resource_util.js:
```diff
@@ -413,3 +401,19 @@ cls.ResourceUtil.http_status_codes = {
+
+  if (reference.protocol == url.protocol)
+  {
+    if (reference.host == url.host)
```
```
if (a || b)
  return true;
```

hzr commented:
After fixing the outstanding issues, this is good to go IMO.

Commit:
Review fixes

Commit:
More specific class name for the link to the resource in the detail view

Commit:
Refactoring of the visibility of the resources

Commit:
Scroll highlighted resource into view

Commit:
Moved the metadata logic of cls.ResourceInfo to a getter

Commit:
Cleaned the scroll into view of the highlighted resource

Commit:
Review fixes

Commit:
Simplified ( the overcomplicated ) logic matching resources to their document

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+			var pos_first = r.url.indexOf(search) - URL_MATCH_CONTEXT_SIZE;
+			var pos_last = r.url.lastIndexOf(search) + URL_MATCH_CONTEXT_SIZE + search.length;
+			var preffix = pos_first > 0 ? "…" : "";
+			var suffix = pos_last < r.url.length ? "…" : "";
```
Could this be done with ellipsis in css instead? Assuming this is to shorten the string for the UI.

p01 replied:
Doing this in CSS would add two class names ( one for prefix ellipsis, one for suffix ellipsis ). Not sure what's best.

Lighter markup and CSS _or_ heavier but more "semantic" CSS and markup

danfooo replied:
It's more flexible when it comes to resizing, and we use that in many other places. Maybe it also solves the searching issues, when the match would be in the shortened part?

Commit:
Added empty setter for cls.ResourceInfo.prototype.metadata

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+		var name = url.filename || url.short_distinguisher || url;
+
+		if (name.length > this.DISTINGUISHER_MAX_LENGTH)
+			name = name.slice(0, this.DISTINGUISHER_MAX_LENGTH) + "…";
```
Here too, not possible to shorten via CSS? Also why would you choose .filename over .short_distinguisher? Also the function name should not claim to just `get_short_distinguisher` then.

p01 replied:
Good point about the `.filename` and co. The `.short_distinguisher` handles the various cases nicely.

As for the ellipsis, again I tend to think that less markup and CSS is better esp. when we are talking about one character.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+
+		this.windows(context);
+		var tpl = ["div", ["ul", this.flat_list], "class", "resource-tree"];
+		delete this.flat_list;
```
Why is `flat_list` not just a var?

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+		if (resources.length > 0)
+		{
+			var depth = d.depth;
+			var extras = this._expander_extras(context, d.pivot_id, depth);
```
Can use `d.depth` directly, `depth` isn't used again.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+						["span",
+							this._get_short_distinguisher(d.url),
+							"class", "resource-tree-document-label",
+							"data-tooltip", " js-script-select",
```
Is the whitespace before `js-script-select` intended?

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+				].concat(extras.tpl.li)
+			);
+
+			if (!extras.collapsed )
```
Extra whitespace after `.collapsed`

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+
+	this.resource_groups = function(context, resources, d)
+	{
+		context.group_order.forEach(this.resource_group.bind(this, context, resources, d));
```
A bit opaque when templates don't return to each other but work on something like `.flat_list` instead. Not sure we ever do that elsewhere. Returning the group_order mapped with something like `resource_group` could work I guess.

p01 replied:
I know.
Flattening the nested return values of these recursively called templates was even more opaque. Ridiculously opaque for what it is.

danfooo replied:
Maybe add a comment when calling forEach about what it will modify. Because it's so unusual in templates.

p01 replied:
Comment added before `this.windows`, line 67

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+		if (!resource.data)
+			return this.no_data_available(resource);
+
+		var specific_template = this[resource.type] ? resource.type : "text";
```
I think something like `var type` would be a better name, since it's also used in the classname.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+	this.overview = function(resource)
+	{
+		var info =
+		{
```
Should go to prev line.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+	{
+		var info =
+		{
+			"response_code": resource.responsecode + " " + cls.ResourceUtil.http_status_codes[resource.responsecode],
```
Looks like it might print `undefined`, maybe should have a check and fall back to "".

p01 replied:
See the beginning of `window.templates.resource_detail.update` ( around the line 230 ):  This template is only called if the corresponding `resourceInfo` object has a `data` property ( i.e. the network entry has a `responsebody` )

danfooo replied:
Ah, I mean the `http_status_codes`. If you say that can't be missing then it's good too.

p01 replied:
Also this is only displayed if the resourceInfo is erroneous according to its responsecode.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+			"character_encoding": resource.encoding || resource.data.characterEncoding
+		};
+
+		var is_error = resource.responsecode && ![200, 304].contains(resource.responsecode);
```
The resourceinfo could copy `error_in_current_response` over, so it would use the same condition.

danfooo commented on src/resource-manager/resource_templates.js:
```diff
@@ -1,6 +1,419 @@
+		return(
+		["object",
+			["div",
+				"Type not supported"
```
Should be a ui_string.

p01 replied:
This is the fallback content of the `object`. I guess I can leave this to the browser and skip my fallback altogether.

rchl replied:
Browser won't show any fallback if you don't specify one. But regardless, You must not embed plugins inside Dragonfly - this opens up for security issues.

danfooo commented on src/resource-manager/resource_tree_view.js:
```diff
@@ -0,0 +1,212 @@
+
+  this._loading = false;
+
+  this.service = resource_inspector;
```
I think this is already renamed in other places?

danfooo commented on src/resource-manager/resource_tree_view.js:
```diff
@@ -0,0 +1,212 @@
+    {
+      container.clearAndRender(tpl);
+      this.tpl_JSON = tpl_JSON;
+    }
```
That's nice, making the tpl and stringifying it is probably really quick? Maybe you can even `return` in else?

p01 replied:
`JSON.stringify(tpl);` is quick! Real quick! I could barely profile it with a nested template around 200 k. Now with the flat list the template is much smaller.

Correct. We can `return` right away No need to fiddle with the scrollTop and scrollLeft in that case.

danfooo commented on src/resource-manager/resource_tree_view.js:
```diff
@@ -0,0 +1,212 @@
+
+    var doc_service = window.services["document-manager"];
+    doc_service.add_listener("abouttoloaddocument", this._on_abouttoloaddocument_bound);
+    doc_service.add_listener("documentloaded", this._on_documentloaded_bound);
```
The is-loading template is probably shown only really short, right? That's all this does? Maybe it's not needed, since as soon as you have even only one resource you'll see something. May be less flashy.

p01 replied:
True. Add the throttling in that picture and there is even less chances of seeing this template. I will nuke it.

Edit: I just nuked it locally and didn't notice any visual glitch. Good call!

danfooo commented on src/network/network_service.js:
```diff
@@ -1293,12 +1289,28 @@ cls.NetworkLoggerResponse.prototype = new cls.NetworkLoggerResponsePrototype();
+cls.ResourceInfo.prototype.__defineGetter__("metadata", function()
+{
+  if (this._metadata === undefined && this.data != null)
+    this._metadata = cls.ResourceUtil.get_meta_data(this);
```
You get a brandnew `ResourceInfo` object on updates, right? Was wondering if that caching could be too agressive, but I guess not.

p01 replied:
Correct.
Also, ideally we should not create a new ResourceInfo object if the corresponding networkLoggerEntry hasn't changed.

danfooo commented on src/scripts/uri.js:
```diff
@@ -53,6 +53,7 @@ var URIPrototype = function(uri_prop_name)
     {
       if (!this._is_data_uri)
       {
+        this.pathname;
```
What does that do?

p01 replied:
o_O nothing.

danfooo commented on src/scripts/uri.js:
```diff
@@ -302,11 +305,16 @@ var URIPrototype = function(uri_prop_name)
+      this._is_parsed = true;
     }
 
-    this._is_parsed = true;
```
What required changing this?

p01 replied:
It was required because I had an object extending URI that didn't get its URL property set in the constructor. The URI class tripped over this pattern. This approach is much cleaner.

Commit:
Review fixes
* no "loading" template
* no "flash" template but a fallback one with a Show Resource link
* reuse the NetworkEntry' .error_in_current_response
* s/service/_resource_inspector in cls.ResourceTreeView and cls.ResourceDetailView
* switch this.flat_list to a private variable in window.templates.resource_tree

Commit:
Added image/webp, show the full URL in the tooltip of the resource, fixed a snafu in the token filtering of the resources

Commit:
Fix for DFL-3668, Esc key does not work

Commit:
Updated string description

Commit:
Bring back the resource's pivot_id which is necessary for the keyboard navigation when closing a group of resources.

Commit:
Fixed detection of relative urls

hzr commented:
Latest commits look fine, just one question.

Commit:
Review fix.
I blame Sublime Text 2 and its crazy contextual modes on that one.

Commit:
Populate the cls.ResourceInfo.data for data: URI resources

Commit:
Display the protocol as fall back origin for cross origin resources without a host ( e.g. data: uri )

Commit:
Added the "text/plain" fall back mimetype and decodeURICompent() as per RFC 2397
Fixed the ellisping of not so short distinguishers

Commit:
DFL-3675: Setting breakpoints from JS source tooltip is broken, sets them in the wrong place

Commit:
Merge pull request #114 from operasoftware/DFL-3675

DFL-3675: Setting breakpoints from JS source tooltip is broken, sets the...

hzr commented:
Looking fine.

Commit:
Merge pull request #115 from operasoftware/escape-control-characters-reviewed

Escape control characters reviewed

Commit:
Merge pull request #116 from operasoftware/bugfixes

Bugfixes

Commit:
Merge pull request #118 from operasoftware/DFL-3602_translations2

Dfl 3602 translations2

Commit:
Merge branch 'cutting-edge' of github.com:operasoftware/dragonfly into cutting-edge

Commit:
Merge branch 'master' of github.com:operasoftware/dragonfly

Commit:
Merge branch 'cutting-edge'

Commit:
Review fixes

Commit:
Synched with master.

Commit:
Declared the required_services for the ResourceTree view
Fixed a problem where keyboard navigation was ignoring the token filter.
Ran cleanrepo

Commit:
Dead code is dead.

Commit:
Fixed greedy filtering of the matching resources in sub documents.
( The children documents of a document were skipped if none of its resource matched the search_term )

Commit:
Fixed infinite search-createView loop and cleaned up the throttling

danfooo commented:
Checked the last 5 commits, seems fine. If it's all reviewed now it would be good to merge it.
